src/main/java/edu/cornell/kfs/core/framework/persistence/ojb/CuStatementManager.java

```diff
@@ -0,0 +1,30 @@
+package edu.cornell.kfs.core.framework.persistence.ojb;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+import org.apache.ojb.broker.PersistenceBroker;
+import org.apache.ojb.broker.accesslayer.StatementManager;
+import org.apache.ojb.broker.metadata.ClassDescriptor;
+import org.apache.ojb.broker.query.Query;
+
+/**
+ * Custom StatementManager subclass that supports param binding for parameterized SQL queries/sub-queries.
+ */
+public class CuStatementManager extends StatementManager {
+
+    public CuStatementManager(final PersistenceBroker pBroker) {
+        super(pBroker);
+    }
+
+    @Override
+    public int bindStatement(PreparedStatement stmt, Query query, ClassDescriptor cld, int param) throws SQLException {
+        int newIndex = param;
+        if (query instanceof QueryByPreparedSQL) {
+            QueryByPreparedSQL preparedQuery = (QueryByPreparedSQL) query;
+            newIndex = bindValues(stmt, preparedQuery.getParameters(), newIndex);
```

Comment:
What is going on with this function, please explain

Reply:
This line is calling a superclass function that binds the given parameters to the prepared SQL statement, using appropriate parameter indexing and returning the next index to use for subsequent param binds. Please let me know if you think this needs more comments or refactoring to explain its intent further.

Reply:
I think some refactoring would be helpful.  I was wondering what this provided base library function didn't